gwcs/coordinate_frames.py

```diff
@@ -731,8 +731,8 @@ def from_index(cls, indexes):
         """
 
         nans = np.isnan(indexes)
-        indexes = np.asanyarray(indexes, dtype=int)
-        out = np.empty_like(indexes, dtype=object)
+        indexes = np.asarray(indexes, dtype=int)
```

Comment:
The bug here was if indexes was a Quantity then it created a quantity for the strings which didn't work.